### ISSUE TITLE
Python3.12 warnings about Invalid escape sequences

### ISSUE THREAD

**Issue by davmlaw** [SACGF/variantgrid#1112]
Invalid escape sequences were [Deprecated in 3.6](https://docs.python.org/dev/whatsnew/3.6.html#deprecated-python-behavior) now Python 3.12  seems to get a bit angry about it, giving warnings:

```
/opt/variantgrid/genes/hgvs/hgvs_converter.py:62: SyntaxWarning: invalid escape sequence '\d'
  if re.match(".*ins\d+$", hgvs_string):
/opt/variantgrid/classification/models/condition_text_matching.py:717: SyntaxWarning: invalid escape sequence '\d'
  pattern = f"{matched_term.id.lower()}(.*?)({prefix}:\d+)"
/opt/variantgrid/uicore/templatetags/js_tags.py:59: SyntaxWarning: invalid escape sequence '\`'
  text = text.replace('\\', '\\\\').replace('`', '\`').replace('</script>', '<\\/script>')
/opt/variantgrid/seqauto/illumina/samplesheet.py:14: SyntaxWarning: invalid escape sequence '\='
  ILLEGAL_CHARACTERS = "?()[]/\=+<>:;\"',*^|&. "  # From Illumina docs
/opt/variantgrid/upload/views/views.py:204: SyntaxWarning: invalid escape sequence '\.'
  accept_file_types = f"/(\.|\/)({'|'.join(extensions)})$/i"
```

This is probably just a matter of adding an "r" to the start, but need to test

**Comment by davmlaw** [comment]
These were all straight forward regex except the [js_tags change](https://github.com/SACGF/variantgrid/commit/08de04213536dafa98d95a971e6aace6e3e683e0#diff-ca48a5865b6e3963019810ced71fb1e814762bc4a9a72bdbf4d82df55e5d5319) which I am 95% sure is right - @TheMadBug - can you make sure that is good?

**Comment by TheMadBug** [comment]
@davmlaw yup, I ran that in iPython and can confirm that the new way is right (and I'm surprised the old way even ran on old Python)

**Comment by davmlaw** [comment]
internal change only - not user testable

**Comment by EmmaTudini** [comment]
Re-opening as can be tested in Shariant - 
This is in relation to finding OMIM, MONDO, PMID etc terms in text. The change is to be explicit in code about special characters in the search pattern (something that older Python’s would allow by default but newer versions wanted you to be explicit about). Can test by making sure citations still work, but pretty safe change.
